packages/mui-material/src/TextField/TextField.js

```diff
@@ -189,7 +189,20 @@ const TextField = React.forwardRef(function TextField(inProps, ref) {
       {...other}
     >
       {label != null && label !== '' && (
-        <InputLabel htmlFor={id} id={inputLabelId} {...InputLabelProps}>
+        <InputLabel
+          htmlFor={id}
+          id={inputLabelId}
+          sx={{
+            ...(select && {
+              // avoid when input is focused or filled
+              [`&:not(.${inputLabelClasses.shrink})`]: {
+                // issue with label padding, see: https://github.com/mui/material-ui/issues/31145
+                maxWidth: variant !== 'standard' ? 'calc(100% - 48px)' : 'calc(100% - 24px)',
+              },
+            }),
+          }}
+          {...InputLabelProps}
+        >
```

Comment:
I don't think this is a good idea. Is it possible to fix the style on the `Select` component instead of the `TextField`?

Reply:
@siriwatknp I think I know why you are suggesting this. Is it because `TextField` is a wrapper of all components and `Select` is a lower-level component and thus it would be better to fix in `Select` component itself?

But with Select component the `InputLabel` is rendered by the user in the form control. The user will have the option to add this custom style if needed. Not for `TextField` with `select` prop.

Reply:
> Is it because TextField is a wrapper of all components and Select is a lower-level component and thus it would be better to fix in Select component itself?

Yes, that is one thing. The other is that `sx` should not be used internally, the styles should be inside a styled component. Something like this:

```js
// TextField.js
const TextFieldRoot = styled(FormControl, {
  name: 'MuiTextField',
  slot: 'Root',
  overridesResolver: (props, styles) => styles.root,
})(({ ownerState }) => ({
  [`& .${inputLabelClasses.root}:not(${inputLabelClasses.shrink})`]: {
    // issue with label padding, see: https://github.com/mui/material-ui/issues/31145
    maxWidth: variant !== 'standard' ? 'calc(100% - 48px)' : 'calc(100% - 24px)',
  }
}));
```

I am still not sure if `TextFieldRoot` is the proper place to own this styles but I will leave that to you.

Reply:
@siriwatknp I have moved it within `TextFieldRoot` styled component. 

> I am still not sure if TextFieldRoot is the proper place to own this styles but I will leave that to you.

I checked and it does work properly. But if you think the solution is an overkill, we can ignore the issue. I'll actually leave it to you.